client/src/dashboard/Assets.vue

```diff
@@ -168,6 +171,26 @@ function openContextMenu(event: MouseEvent, key: AssetId): void {
 
 let draggingSelection = false;
 
+function renameAsset(event: FocusEvent, file: AssetId, oldName: string): void {
+    if(!canEdit(file, false)) {
+        return;
+    }
+
+    const target = event.target as HTMLElement;
+    if(target.textContent === null){
+        target.textContent = oldName;
+        currentRenameAsset.value = null;
+        return;
+    }
+    const name = target.textContent.trim();
+
+    if (name === undefined || name === "" || name === "..") {
+        target.textContent = oldName;
+    }else{
```

Comment:
I'm surprised the linter doesn't pick this up, but add some spacing around the else.

Maybe also make it an `else if (name !== oldName)`. The extra network call isn't going to be a big issue, but nevertheless we know it's not necessary.

Reply:
Makes sense, fixed!